gui/searchFeature.js

```diff
@@ -3,24 +3,33 @@ let searchResults = document.getElementById('searchResults');
 
 searchValue.addEventListener('keyup', () => {
   let searchPhrase = searchValue.value.trim();
-  if (searchPhrase.length < 1) { return; }
+  if (searchPhrase.length < 1) {
+    searchResults.innerText = '';
+    return;
+  }
 
   searchPhrase = searchPhrase.toLowerCase();
 
-  let verses = '';  
+  let verses = '';
+  let foundResults = false;
 
   for (let i = 1; i <= 18; i++) {
     for (let j = 1; j <= Object.keys(BHAGAVAD_GITA[i]).length; j++) {
       let sanskritVerse = BHAGAVAD_GITA[i][j][0];
       let englishVerse = BHAGAVAD_GITA[i][j][1];
 
       if (englishVerse.toLowerCase().includes(searchPhrase)) {
-        verses += 
+        foundResults = true;
+        verses +=
           "Chapter " + i + ", Verse " + j + "\n" +
-          sanskritVerse + "\n" + englishVerse + "\n\n"
+          sanskritVerse + "\n" + englishVerse + "\n\n";
       }
     }
   }
 
-  searchResults.innerText = verses;
-})
\ No newline at end of file
+  if (!foundResults) {
```

Comment:
Small tip for readability:

Keep this line as if (foundResults)
And then swap lines 31 with 33.